cmakelib/lexer/lexer.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package lexer implements a participle Lexer for the CMakeLists.txt language.
+// See https://cmake.org/cmake/help/v3.0/manual/cmake-language.7.html for more details.
+package lexer
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/alecthomas/participle/lexer"
+)
+
+const (
```

Comment:
I'm a little surprised golint doesn't complain about the lack of comments on these exported names. You can appease it by putting a comment on the block, though, e.g.,

`// Constants defining the token types used by CMake`

or words to that effect.

Reply:
I haven't hooked up golint in this repo yet is why it doesn't complain. Thankfully, this appears to be its only complaint :-)